File: Domain/Message.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Domain
{
    public class Message
    {
        public int MessageId { get; set; }
        public AppUser Sender { get; set; }
        public string Content { get; set; }
        public DateTime SentAt { get; set; }
        public bool User1Deleted { get; set; }
        public bool User2Deleted { get; set; }
        public bool IsRead { get; set; }
        public int ConversationId { get; set; }
        public virtual Conversation Conversation { get; set; }

    }
}

File: client-app/src/app/models/message.ts
import { User } from "./user"

export interface IMessage {
    messageId: number;
    sender: User;
    content: string;
    sentAt: Date;
    isRead: boolean;
    conversationId: number
    //'User1Deleted': boolean;
    //'User2Deleted': boolean;
}

export class Message implements IMessage {
    constructor(messageId: number, content: string, sentAt: Date, isRead: boolean) {
        this.messageId = messageId;
        this.content = content;
        this.sentAt = sentAt;
        this.isRead = isRead;
    }
    messageId: number;
    sender: User = new User();
    content: string;
    sentAt: Date;
    isRead: boolean;
    conversationId = 0;
}